File: Item.py
import Producto


class Item:

    def __init__(self, producto, cantidad):
        self.producto = producto
        self.cantidad = cantidad

    def calcular_total(self):
        manejador_reglas = ManejadorReglas()
        if self.producto.codigo_sku.startswith("EA"):
            manejador_reglas.set_regla(ReglaNormal())
        elif self.producto.codigo_sku.startswith("WE"):
            manejador_reglas.set_regla(ReglaPeso())
        elif self.producto.codigo_sku.startswith("SP"):
            manejador_reglas.set_regla(ReglaEspecial())

        return manejador_reglas.ejecutar_regla(self.cantidad, self.producto.precio_unitario)

File: Backend/model/Item.ts
import ManejadorReglas from "./ManejadorReglas";
import Producto from "./Producto";

export default class Item{
    private producto: Producto;
    private cantidad: number;
    constructor(producto: Producto, cantidad: number){
        this.producto = producto;
        this.cantidad =  cantidad;
    }

    public getProducto(): Producto{
        return this.producto;
    }

    public getCantidad(): number{
        return this.cantidad;
    }

    public calcularTotal(): (number | undefined){
        const manejadorReglas =  new ManejadorReglas();
        return manejadorReglas.ejecutarRegla(this.cantidad, this.producto.precioUnitario, this.producto.sku);
    }
}